tests/ssl-opt.sh

```diff
@@ -1112,6 +1112,103 @@ run_test    "Truncated HMAC, DTLS: client enabled, server enabled" \
             -S "dumping 'expected mac' (20 bytes)" \
             -s "dumping 'expected mac' (10 bytes)"
 
+# Tests for DTLS Connection ID extension
+
+# TODO
+# So far, the CID API isn't implemented, so we can't
+# grep for output witnessing its use. This needs to be
+# changed once the CID extension is implemented.
```

Comment:
[I tried leaving that comment before but now I can't see it anywhere, so leaving it again] I'd much rather have the tests be complete and skipped than incomplete, for two reasons:

1. Incomplete tests diminishes the value of writing them and reviewing them now.
2. As long as the feature isn't actually implemented, I don't think tests should print "PASS" as that would be misleading - "SKIP" on the other hand is much more truthful.

Reply:
I am not sure about this, because we can gradually make the tests more expressive. Specifically: (a) Once the new CID handshake fields are in place and configurable through the top-level API, their setting can be witnessed by a debug message we can grep for. (b) Once the CID extension writing/parsing is in place, we can witness that in a debug message we can grep for. (c) Once the CID record protection is in place, we can witness that in a debug message we can grep for.
So, in summary, I expect more and more debug output to be grepped for here, and already the intermediate stages have value.

Reply:
You're making excellent points here, I hadn't considered that strategy.

I'm still worried that the output of the script might be misleading. Can we change the description of individual tests then, adding something like "STUB" or "WORK IN PROGRESS - extension sent and received" (then update that at each step until we can remove the WIP part altogether)?

Reply:
Good idea, I'll do that.

Reply:
Done.